### PR TITLE
Add --json flag for json output/input

### PR DESCRIPTION
**Pull request by esbie**
It's easier to use namerctl programmatically if the format is json. This adds the --json flag for dtab commands.

### CODE REVIEW

**Comment by esbie** [comment]
My preference is to review & merge this with failing tests, since the act of merging to master will resolve the issue

File: cmd/dtab.go [hunk 1]
```diff
@@ -93,7 +107,7 @@ var (
 				if err != nil {
 					return err
 				}
-				_, err = ctl.Create(name, dtabstr)
+				_, err = ctl.Create(name, dtabstr, dtabJson)
```
Comment (klingerf) [inline cmd/dtab.go]: Hmm, not sure what I am doing wrong locally, but I couldn't get `create --json` to work.

```
$ cat test.dtab 
{"dtab": [{"prefix":"/http","dst":"/$/inet/veggies/7003"}]}
$ go run main.go dtab --base-url http://192.168.99.100:4180 --json create newns test.dtab 
Error: unexpected response: 400 Bad Request
```

File: cmd/dtab.go [hunk 2]
```diff
@@ -124,7 +138,7 @@ var (
 				if err != nil {
 					return err
 				}
-				_, err = ctl.Update(name, dtabstr, namer.Version(dtabUpdateVersion))
+				_, err = ctl.Update(name, dtabstr, dtabJson, namer.Version(dtabUpdateVersion))
```
Comment (klingerf) [inline cmd/dtab.go]: Same goes for update as well.  Maybe I don't have the right version of namerd?  I just re-pulled the nightly docker image.

Comment (esbie) [inline cmd/dtab.go]: It looks like I'm use namerd v7, so you're right there could be an incompatibility here https://github.com/BuoyantIO/linkerd-examples/blob/master/gob/k8s/namerd/rc.yml#L24

File: cmd/dtab.go (hunk 1 above)
Comment (esbie) [inline cmd/dtab.go]: The problem is I forgot to update these commands to support the `{"dtab": ..., "version" ...}` structure (whereas using the dtab array directly works fine). I will post a fix thank you!

File: namer/controller.go [hunk 3]
```diff
@@ -47,42 +47,53 @@ func (ctl *httpController) dtabRequest(method, name string, data io.Reader) (*ht
...
 	req, err := ctl.dtabRequest("GET", name, nil)
 	if err != nil {
-		return nil, err
+		return nil, "", err
+	}
+
+	if requestJson {
+		req.Header.Set("Accept", "application/json")
+	} else {
+		req.Header.Set("Accept", "application/dtab")
```
Comment (olix0r) [inline namer/controller.go]: This seems funny to me.  Should we just be able to JSON-encode a VersionedDtab?

**Comment by esbie** [comment]
@olix0r sure, we can move to always talking to the json api everywhere. It irks me to have to parse the json response only to re-marshal it for the CLI, which is why I avoided it.

**Pull request merged**